devices/xiaomi.js

```diff
@@ -211,16 +242,84 @@ const tzLocal = {
+                let repeatRule = 254;
+                for (let i = 1; i <= 7; i++) {
+                    if (value.repeat.some((e)=>e.toUpperCase() == dayNames[i-1].toUpperCase()) == false) {
+                        repeatRule = repeatRule & ~(1<<i);
+                    }
+                }
+                const payload = [
+                    0x04,
+                    repeatRule,
+                ];
+
+                const getValue = (time, temperature) => {
+                    const val = Buffer.alloc(6);
+                    let totalMinutes = time.hours * 60 + time.minutes;
+                    if (time.nex_day) {
+                        totalMinutes = totalMinutes | 1<<15;
+                    }
+                    val.writeUint16BE(totalMinutes);
+                    val.writeUint16BE(temperature*100, 4);
+                    return val;
+                };
+                const getTime = (time) => {
+                    let totalMinutes = time.hours * 60 + time.minutes;
+                    if (time.nex_day) {
+                        totalMinutes = totalMinutes + 1440;
+                    }
+                    return totalMinutes;
+                };
+
+                for (let i = 0; i < 3; i++) {
+                    const startTime = getTime(value.intervals[i].from);
+                    const endTime = getTime(value.intervals[i].to);
+
+                    if (i !=0 && getTime(value.intervals[i-1].to) != startTime) {
+                        throw new Error(`The start time of the interval must be equal to the end time of the previous interval;
+                        (index: ${i})`);
+                    }
+
+                    if ((endTime - startTime) < 60) {
+                        throw new Error(`The end time must be greater than the start time, minimum duration is one hour;(index: ${i})`);
+                    }
+
+                    const temperature = value.intervals[i].temperature;
+                    payload.push(...getValue(value.intervals[i].from, temperature));
+
+                    if (i == 2) {
+                        payload.push(...getValue(value.intervals[i].to, temperature));
+                    }
+                }
+
+                await entity.write('aqaraOpple', {0x0276: {value: payload, type: 0x41}}, {manufacturerCode: 0x115f});
+                break;
+            }
             default: // Unknown key
                 meta.logger.warn(`zigbee-herdsman-converters:aqara_trv: Unhandled key ${key}`);
             }
         },
         convertGet: async (entity, key, meta) => {
-            const dict = {'system_mode': 0x0271, 'preset': 0x0272, 'window_detection': 0x0273, 'valve_detection': 0x0274,
-                'child_lock': 0x0277, 'away_preset_temperature': 0x0279, 'calibrated': 0x027b, 'sensor': 0x027e};
+            const dict = {'state': [0x0271], 'preset': [0x0272], 'window_detection': [0x0273], 'valve_detection': [0x0274],
```

Comment:
Shouldnt 'state' here be 'system_mode', see the last merged PR. Think youve conflicted here